docker-app/qfieldcloud/core/management/commands/extractstoragemetadata.py

```diff
@@ -0,0 +1,133 @@
+import csv
+import logging
+import sys
+from typing import Generator, NamedTuple
+
+import boto3
+import mypy_boto3_s3
+from django.core.management.base import BaseCommand, CommandParser
+from qfieldcloud.core import utils
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """Save metadata from S3 storage project files to disk."""
+
+    def add_arguments(self, parser: CommandParser):
+        parser.add_argument(
+            "-o", "--output", type=str, help="Optional: Name of output file"
+        )
+        parser.add_argument(
+            "-s3",
+            "--s3_credentials",
+            type=str,
+            help="""
+            Optional: S3 credentials as comma-separated key=value pairs.
+            Example: -s3 STORAGE_ACCESS_KEY_ID=...,STORAGE_SECRET_ACCESS_KEY=...,...
+        """,
+        )
```

Comment:
Please add all the credentials as separate arguments. It is more explcit and will reduce your code in terms of size and complexity, e.g. `from_shell` will become obsolete.